### DISCUSSION TITLE
Email flows / OAuth client / passwordless

### DISCUSSION THREAD

**Discussion by tennox** [kanidm/kanidm#2327] [Q&A] [answered]
Hi there :)
Just starting to check out the project, and liking many parts of it! :heart_eyes: :crab: 

I'm realizing though, it seems to not be quite usable for my current case, as we have a lot of old / non-techsavvy users, and Password/TOTP isn't really an option. (And webauthn not widely supported afaik).

What about OAuth Client? ("Sign up with google") - I don't like it, but for many users it would make life a lot easier. :thinking: 
Can understand if it's not necessarily the route you want to go down.

I'm also wondering what your plans are regarding integration with common email-based flows, like:
- reset password (found an [open issue](https://github.com/kanidm/kanidm/issues/2299), but not mentioning emails)
- sign-up ([open issue](https://github.com/kanidm/kanidm/issues/2287) mentions the )
- passwordless login

Basically just checking options here, if this project might somehow work for us - because I would like it to :sweat_smile: 

I'm ready and able to implement some parts myself, but only in smaller scope for now. :tipping_hand_man:

**Comment by yaleman** [comment]
> I'm realizing though, it seems to not be quite usable for my current case, as we have a lot of old / non-techsavvy users, and Password/TOTP isn't really an option. (And webauthn not widely supported afaik).

If they can't use password/totp or webauthn (which works in every modern browser/platform) what options for secure authentication are left?

> What about OAuth Client? ("Sign up with google") - I don't like it, but for many users it would make life a lot easier. 🤔

IMO, we're the thing you'd sign up with 😄 but you could build your own cross-connect.

**Comment by Firstyear** [comment, marked as answer]
> Hi there :) Just starting to check out the project, and liking many parts of it! 😍 🦀

Thanks for the really nice words :) 

> 
> I'm realizing though, it seems to not be quite usable for my current case, as we have a lot of old / non-techsavvy users, and Password/TOTP isn't really an option. (And webauthn not widely supported afaik).

Webauthn is supported on every browser and iphone today (android is the key weakness here, which has huge adoption issues). 

> 
> What about OAuth Client? ("Sign up with google") - I don't like it, but for many users it would make life a lot easier. 🤔 Can understand if it's not necessarily the route you want to go down.

This kind of defeats the point of Kanidm then doesn't it? If you want an external oauth provider, then why use Kanidm at all? 

> 
> I'm also wondering what your plans are regarding integration with common email-based flows, like:
> 
>     * reset password (found an [open issue](https://github.com/kanidm/kanidm/issues/2299), but not mentioning emails)
> 
>     * sign-up ([open issue](https://github.com/kanidm/kanidm/issues/2287) mentions the )
> 
>     * passwordless login
> 
> 
> Basically just checking options here, if this project might somehow work for us - because I would like it to 😅
> 
> I'm ready and able to implement some parts myself, but only in smaller scope for now. 💁‍♂️

Most of these "ideas" are quite large to implement I think. 

We aren't really interested in "passwordless" login (aka email token sending), and to implement it would be a large amount of work IMO. 

Sign-up/reset is kind of on the todo, but it's a longer term goal - our current goals are other things right now.
